pkg/cmd/server/api/helpers.go

```diff
@@ -120,6 +121,10 @@ func GetMasterFileReferences(config *MasterConfig) []*string {
 	refs = append(refs, &config.ServingInfo.ServerCert.CertFile)
 	refs = append(refs, &config.ServingInfo.ServerCert.KeyFile)
 	refs = append(refs, &config.ServingInfo.ClientCA)
+	for i := range config.ServingInfo.NamedCertificates {
```

Comment:
> I only added references for the ones we support (I explicitly disallow it in etcdConfig via validation, since etcd doesn't support SNI)... add to HTTPServingInfo instead? felt weird there

I'm just looking at this function and decision about file ref or not is based mostly on type.  I'd feel better about finding all references to types and adding them as file refs.